File: src/modifiers/accessibilityHint.ts
import { createModifier } from './createModifier';

/// SwiftUI `.accessibilityHint(_:)`. Describes the result of performing the
/// view's action, read by VoiceOver after the label.
export function accessibilityHint(hint: string) {
  return createModifier('accessibilityHint', { hint });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/AccessibilityHint.swift
import SwiftUI

/// SwiftUI `.accessibilityHint(_:)`. Describes the outcome of the view's
/// action; read by VoiceOver after the label.
enum RNWAccessibilityHintModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("accessibilityHint") { view, params, _ in
            AnyView(view.accessibilityHint(Text(params.string("hint") ?? "")))
        }
    }
}
